terraform/install.sh

```diff
@@ -256,28 +258,30 @@ getExternalIp() {
 # Install Load Generator service and start generating synthetic traffic to Sandbox
 loadGen() {
   log "Running load generator"
-  # remove existing load generator
-  old_loadgen=$(gcloud compute instances list --project $project_id --filter="name:loadgenerator*" --format="value(name)")
-  if [[ -n "$old_loadgen" ]]; then
-    loadgen_zone=$(gcloud compute instances list --project $project_id --filter="name:loadgenerator*" --format="value(zone)")
-    gcloud compute instances delete $old_loadgen --zone $loadgen_zone --quiet
-  fi
   # launch a new load generator
-  ../loadgenerator/loadgen autostart $external_ip
+  pushd loadgen/
+  terraform init -lock=false
+  terraform apply --auto-approve -var="project_id=${project_id}" -var="external_ip=${external_ip}"
+  popd
+
   # find the IP of the load generator web interface
   TRIES=0
-  while [[ $(curl -sL -w "%{http_code}"  "http://$loadgen_ip:8080" -o /dev/null --max-time 1) -ne 200  && \
+  while [[ $(curl -sL -w "%{http_code}"  "http://$loadgen_ip:$LOCUST_PORT" -o /dev/null --max-time 1) -ne 200  && \
       "${TRIES}" -lt 20  ]]; do
     log "waiting for load generator instance..."
-    sleep 1
-    loadgen_ip=$(gcloud compute instances list --project "$project_id" \
-                                               --filter="name:loadgenerator*" \
-                                               --format="value(networkInterfaces[0].accessConfigs.natIP)")
+    sleep 10
+    loadgen_ip=$(kubectl get service locust-main -o jsonpath='{.status.loadBalancer.ingress[0].ip}');
+     [ -z "$loadgen_ip" ] && sleep 10;
     TRIES=$((TRIES + 1))
   done
-  if [[ $(curl -sL -w "%{http_code}"  "http://$loadgen_ip:8080" -o /dev/null  --max-time 1) -ne 200 ]]; then
+  if [[ $(curl -sL -w "%{http_code}"  "http://$loadgen_ip:$LOCUST_PORT" -o /dev/null  --max-time 1) -ne 200 ]]; then
```

Comment:
nit: I don't like seeing the query duplicated here. It would be great if you could find a way to use the result from above. But I know your time is limited, so don't worry too much about this